src/qibocal/protocols/resonator_spectroscopy.py

```diff
@@ -201,25 +210,46 @@ def _fit(
     error_fit_pars = {}
     chi2 = {}
     amplitudes = {}
+
+    fit = s21_fit if data.fit_function == "s21" else lorentzian_fit
```

Comment:
```suggestion
    fit, chi2_, values, errors = (
        (lorentzian_fit, chi2_reduced, lambda q: q.signal, lambda q: q.error_signal)
        if data.fit_function == "lorentzian"
        else (
            s21_fit,
            chi2_reduced_complex,
            lambda q: (q.signal, q.phase),
            lambda q: (q.error_signal, q.error_phase),
        )
    )
```

After this one, you won't need any further branching :)

Reply:
Together with the `plot` function below, these could even be collected in a data structure, with the two tuples.

E.g. for the time being, you could define something like:
```py
@dataclass
class SpectroscopyFit:
    fit: Callable
    chi2: Callable
    values: Callable
    errors: Callable
    plot: Callable

fits = {"lorentzian": SpectroscopyFit(lorentzian_fit, ...), "s21": SpectroscopyFit(...)}
```
top-level, since the definitions do not depend on any runtime data other than the fit kind.